src/wallet/external_signer_scriptpubkeyman.cpp

```diff
@@ -49,15 +51,26 @@ ExternalSigner ExternalSignerScriptPubKeyMan::GetExternalSigner() {
     return signers[0];
 }
 
-bool ExternalSignerScriptPubKeyMan::DisplayAddress(const CScript scriptPubKey, const ExternalSigner &signer) const
+std::optional<bilingual_str> ExternalSignerScriptPubKeyMan::DisplayAddress(const CTxDestination& dest, const ExternalSigner &signer) const
 {
     // TODO: avoid the need to infer a descriptor from inside a descriptor wallet
+    CScript scriptPubKey = GetScriptForDestination(dest);
     auto provider = GetSolvingProvider(scriptPubKey);
     auto descriptor = InferDescriptor(scriptPubKey, *provider);
 
-    signer.DisplayAddress(descriptor->ToString());
-    // TODO inspect result
-    return true;
+    UniValue result = signer.DisplayAddress(descriptor->ToString());
+
+    const UniValue& error = find_value(result, "error");
+    if (error.isStr()) return strprintf(_("Signer returned error: %s"), error.getValStr());
+
+    const UniValue& ret_address = find_value(result, "address");
+    if (!ret_address.isStr()) return _("Signer did not echo address");
```

Comment:
Won't this break older external signers?

Reply:
HWI has always done this, we just didn't check. That said, it's not (yet) in the spec (which may be a bit outdated) so other signer software might not do this. Do you know of any? cc @achow101 

 https://github.com/bitcoin/bitcoin/blob/master/doc/external-signer.md#displayaddress-optional

Reply:
AFAIK there aren't any other external signer software yet.